backend/src/main/java/gov/cdc/usds/simplereport/api/patientExperience/PatientExperienceController.java

```diff
@@ -0,0 +1,123 @@
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import static gov.cdc.usds.simplereport.api.Translators.parseSymptoms;
+import static gov.cdc.usds.simplereport.api.Translators.parseEmail;
+import static gov.cdc.usds.simplereport.api.Translators.parseEthnicity;
+import static gov.cdc.usds.simplereport.api.Translators.parseGender;
+import static gov.cdc.usds.simplereport.api.Translators.parsePhoneNumber;
+import static gov.cdc.usds.simplereport.api.Translators.parseRace;
+import static gov.cdc.usds.simplereport.api.Translators.parseState;
+import static gov.cdc.usds.simplereport.api.Translators.parseString;
+
+import gov.cdc.usds.simplereport.api.exceptions.FeatureFlagDisabledException;
+import gov.cdc.usds.simplereport.api.model.AoEQuestions;
+import gov.cdc.usds.simplereport.api.model.pxp.PxpApiWrapper;
+import gov.cdc.usds.simplereport.api.model.pxp.PxpPersonWrapper;
+import gov.cdc.usds.simplereport.db.model.Organization;
+import gov.cdc.usds.simplereport.db.model.PatientLink;
+import gov.cdc.usds.simplereport.db.model.Person;
+import gov.cdc.usds.simplereport.db.model.TestEvent;
+import gov.cdc.usds.simplereport.db.model.auxiliary.TestResult;
+import gov.cdc.usds.simplereport.service.PatientLinkService;
+import gov.cdc.usds.simplereport.service.PersonService;
+import gov.cdc.usds.simplereport.service.TestEventService;
+import gov.cdc.usds.simplereport.service.TestOrderService;
+
+@RestController
+@RequestMapping(value = "/pxp")
+@Validated
+public class PatientExperienceController {
+  @Autowired
+  private PersonService ps;
+
+  @Autowired
+  private PatientLinkService pls;
+
+  @Autowired
+  private TestOrderService tos;
+
+  @Autowired
+  private TestEventService tes;
+
+  @Value("${feature-flags.patient-links:false}")
+  private boolean patientLinksEnabled;
+
+  /**
+   * This endpoint is used to determine whether or not the patient link is still
+   * valid and has not yet expired. It does not require birthdate verification
+   */
+  @GetMapping("/link/{plid}")
+  public Organization getPatientLinkCurrent(@PathVariable("plid") String internalId) throws Exception {
+    if (!patientLinksEnabled) {
+      throw new FeatureFlagDisabledException("Patient links not enabled");
+    }
+    return pls.getPatientLinkCurrent(internalId);
+  }
+
+  /**
+   * Verify that the patient-provided DOB matches the patient on file for the
+   * patient link id. It returns the full patient object if so, otherwise it
+   * throws an exception
+   */
+  @PutMapping("/link/verify")
+  public PxpPersonWrapper getPatientLinkVerify(@RequestBody PxpApiWrapper<Void> body) throws Exception {
+    if (!patientLinksEnabled) {
+      throw new FeatureFlagDisabledException("Patient links not enabled");
+    }
+    Person p = pls.getPatientLinkVerify(body.getPlid(), body.getDob());
```

Comment:
Could you catch both `IllegalGraphqlArgumentException` and `IncorrectBirthDateException` here and collapse them into a single error state? This should be something that gets converted to a 401 HTTP status code and have a message like, "No patient link with the supplied ID was found, or the birth date provided was incorrect."

It may be safest to collapse that distinction in PatientLinkService rather than in each controller method, as anything calling `pls.getPatientLinkVerify` needs to be similarly opaque about which factor failed.

Reply:
Good call, yes! Thanks

Reply:
This is resolved [here](https://github.com/CDCgov/prime-simplereport/commit/d328a3650c020f9731f73ac14add88ffbf90296b#diff-6c96fbf1b07aea7f400f136be0ee47610dafd2a9d2a5803d32d467e0f836640f) and [here](https://github.com/CDCgov/prime-simplereport/commit/d328a3650c020f9731f73ac14add88ffbf90296b#diff-62083adfe1a711f8771fb60ed33935ea2658af693757c598565391a50c524fd4]